### PR TITLE
DOCS update LinkingModes description

### PR DESCRIPTION
**Pull request by muskie9**
see silverstripe/silverstripe-cms#1325

### CODE REVIEW

**Comment by dhensby** [comment]
Can you please recommend to use the `is` functions in the template than relying on the returned string from the functions?

**Comment by muskie9** [comment]
Sure can, I saw your comment on the other PR. I probably would've gone with just a docs update initially rather than the cms PR if I had realized those `is` functions existed.

**Comment by muskie9** [comment]
I've updated this to reference the `is` methods. Let me know if it should be adjusted at all.

File: docs/en/02_Developer_Guides/01_Templates/02_Common_Variables.md [hunk 1]
```diff
@@ -184,6 +184,14 @@ For instance, to only show the menu item linked if it's the current one:
 		<p>You are viewing the about us section</p>
 	<% end_if %>
 
+If you need alternative values for for "current" or "section" it's suggested you check against the `is` methods (`$isSection` and `$isCurrent`) and define you'r own value in the template:
```
Comment (dhensby) [inline docs/en/02_Developer_Guides/01_Templates/02_Common_Variables.md]: `you'r` should be your

Comment (jonom) [inline docs/en/02_Developer_Guides/01_Templates/02_Common_Variables.md]: * `for for "current"` should be `for "current"`
* `you'r` should be `your`

File: docs/en/02_Developer_Guides/01_Templates/02_Common_Variables.md [hunk 2]
```diff
@@ -184,6 +184,14 @@ For instance, to only show the menu item linked if it's the current one:
...
+
+    :::ss
+    <a class="
+        <% if $isSection %>my-section-class<% end_if %>
+        <% if $isCurrent %>my-current-class<% end_if %>
+    " href="foo">Foo</a>
```
Comment (jonom) [inline docs/en/02_Developer_Guides/01_Templates/02_Common_Variables.md]: Can you make this `" href="$Link">$MenuTitle</a>`

(sorry I should have included this in the original example)

**Comment by jonom** [comment]
This is a good start and would be a good addition on its own. If you're up for it though @muskie9 think it would be nice to also deprecate the `LinkOrCurrent` and `LinkOrSection` methods in this PR (and possibly the LinkingMode method too) and update all the examples that use those methods to use the `is` methods instead. Then it can be all wrapped up neatly in one commit.

Relevant doc files are:
* 01_Tutorials/01_Building_A_Basic_Site.md
* 02_Developer_Guides/01_Templates/02_Common_Variables.md
* 02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md

I'd be fine with deprecating `LinkingMode` and I think @dhensby is too. @clarkepaul do you have an opinion on this since it's more of a front-end issue?

Background discussion here: https://github.com/silverstripe/silverstripe-cms/pull/1325

**Comment by muskie9** [comment]
I should've proof read my commit a bit more before force pushing I guess ;) I can take a shot at deprecating the methods if that's the consensus. I'm assuming the PR would go against cms:master for the deprecation.

**Comment by dhensby** [comment]
I think avoid the depreciation here. It will block merge and we'll need concensus to take it in.

**Comment by jonom** [comment]
> Minor version Y (x.Y.z | x > 0) ... MUST be incremented if any public API functionality is marked as deprecated

So FYI marking methods deprecated for 4.0 would go in to 3 branch, but @dhensby's got a point so maybe don't do that yet ;)

Still would be cool to update the docs to not mention those string methods if you have time, would highlight their redundancy and pave the way for deprecating them in the future.

**Comment by dhensby** [comment]
Ahreed

**Comment by muskie9** [comment]
:+1:

**Comment by muskie9** [comment]
I've done some preliminary updates to the docs noted above. Let me know if I missed anything or wording needs to be changed.

File: docs/en/02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md [hunk 3]
```diff
@@ -11,15 +11,15 @@ top level menu with a nested second level using the `Menu` loop and a `Children`
 	<ul>
 		<% loop $Menu(1) %>
 			<li>
-				<a href="$Link" title="Go to the $Title page" class="$LinkingMode">
+				<a href="$Link" title="Go to the $Title page" class="<% if $isSection %>section<% end_if %> <% if $isCurrent %>$current<% end_if %>">
```
Comment (jonom) [inline docs/en/02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md]: Extra $ sign in there, should be `<% if $isCurrent %>current<% end_if %>`

File: docs/en/02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md [hunk 4]
```diff
@@ -11,15 +11,15 @@ top level menu with a nested second level using the `Menu` loop and a `Children`
...
 					$MenuTitle
 				</a>
 
-				<% if $LinkOrSection == section %>
+				<% if $isSection %>
 					<% if $Children %>
 						<ul class="secondary">
 							<% loop $Children %>
-								<li class="$LinkingMode"><a href="$Link">$MenuTitle</a></li>
+								<li class="<% if $isSection %>section<% end_if %> <% if $isCurrent %>$current<% end_if %>"><a href="$Link">$MenuTitle</a></li>
```
Comment (jonom) [inline docs/en/02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md]: same here :)

Comment (muskie9) [inline docs/en/02_Developer_Guides/01_Templates/How_Tos/01_Navigation_Menu.md]: Must have been muscle memory lol

**Comment by jonom** [comment]
I think it looks good apart from a couple of stray `$` signs.

Sorry for being so pedantic and changing my mind on my own feedback but I wonder if we should change all these:
```
class="<% if $isSection %>section<% end_if %> <% if $isCurrent %>current<% end_if %>"
```
to 
```
class="<% if $isCurrent %>current<% else_if $isSection %>section<% end_if %>"
```
This then mirrors the $LinkingMode output but without adding the redundant 'link' class, and it looks a bit cleaner. More backwards compatible and a good example for accomplishing your original goal.

**Comment by muskie9** [comment]
@jonom I was actually going to ask if we should go as far as `<% if $isCurrent %>current<% else_if $isSection %>section<% else %>link<% end_if %>`

**Comment by jonom** [comment]
@muskie9 :+1:

p.s. my bad I should have remembered that tutorials are deprecated in favour of lessons. Still worth updating I think though.

@unclecheese what would be involved in updating http://www.silverstripe.org/learn/lessons/adding-dynamic-content to not use $LinkingMode? We're thinking it's easy to accomplish same with $isSection and $isCurrent, so may be better to show developers how to provide their own class names in example. This should be especially handy for people working with frameworks like Bootstrap.

**Comment by jonom** [comment]
> I was actually going to ask if we should go as far as <% if $isCurrent %>current<% else_if $isSection %>section<% else %>link<% end_if %>

Personally I would prefer to leave out the link part... I can't think of anytime the 'link' class would be useful when you can just target the `a` element, so it seems redundant to me.

**Comment by muskie9** [comment]
I've updated the docs per your suggestion @jonom. I don't do a lot of front-end dev, so I presume most links are set via `a` rather than `a.link` in most cases (outside of current and/or section).

**Comment by jonom** [comment]
Thanks for seeing it through @muskie9!

**Comment by muskie9** [comment]
Happy to! Glad we got it figured out!

**Comment by clarkepaul** [comment]
@jonom @muskie9  I haven't really been involved in the frontend side of things for a while now but from what I can tell this sounds like a good idea and I can see it being useful. I've sent a link to few others here just in case there is any concerns.

**Comment by clarkepaul** [comment]
Ah, I didn't see this was merged - no worries, happy days

**Comment by jonom** [comment]
@clarkepaul we haven't deprecated anything yet though so if you have an opinion on that let us know ;)

**Comment by dhensby** [comment]
@clarkepaul this is just a DOCs change now; we haven't made it customisable... yet

**Pull request merged**